packages/vite-plugin-svelte/src/utils/options.ts

```diff
@@ -151,6 +152,45 @@ export async function resolveOptions(
 	return resolvedOptions;
 }
 
+export function buildExtraViteConfig(
+	options: ResolvedOptions,
+	config: UserConfig
+): Partial<UserConfig> {
+	const allSvelteImports = [...SVELTE_IMPORTS, ...SVELTE_HMR_IMPORTS];
+
+	// exclude svelte imports from optimization unless explicitly included
+	const excludeFromOptimize = allSvelteImports.filter(
+		(x) => !config.optimizeDeps?.include?.includes(x)
+	);
```

Comment:
Something I just tested, if the user provide the svelte imports in `optimizeDeps.include` of the config directly, meaning the svelte imports would end up in both `optimizeDeps.include` and `optimizeDeps.exclude`, Vite pre-bundling seems to have `include` take precedence. But I think it's safer this way too.

Just something I found out.

Reply:
yeah, even if include takes precedence, having it in both is confusing.